File: zeroFilledSubarray.py
# Approach 1- Count the no. of consecutive 0's at each index
def zeroFilledSubarray(nums): # TC O(n) // SC O(1)
	ans,zeroSubarray=0,0
	for num in nums:
		if num==0:
			zeroSubarray+=1
		else:
			zeroSubarray=0 
		ans+=zeroSubarray
	return ans 

print(zeroFilledSubarray([1,3,0,0,2,0,0,4]))

File: zeroFilledSubarray.java
import java.util.*;
// Approach 1- Count the no. of consecutive 0's at each index
class Solution { // TC O(n) SC O(1)
    public long zeroFilledSubarray(int[] nums) {
        long ans=0;
        long zeroSubarray=0;
        for(int num: nums){
            if (num==0){
                zeroSubarray++;
            } else {
                zeroSubarray=0;
            }
            ans+=zeroSubarray;
        }
        return ans;
    }
}

class zeroFilledSubarray{
    public static void main(String[] args) {
        int[] nums={1,3,0,0,2,0,0,4};
        Solution s= new Solution();
        System.out.println(s.zeroFilledSubarray(nums));
    }
}